### PR TITLE
Small logging fix, extra SC execution traces

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Fixed (applied upper limit) the `hospitability indicator` (used for logging only). Added duration monitoring for SC execution.
 - Add capacity limit for map of immune keys (current or future items).

### CODE REVIEW

File: process/smartContract/process.go [hunk 1]
```diff
@@ -160,12 +163,42 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
...
-	log.Trace("scProcessor.ExecuteSmartContractTransaction()", "sc", tx.GetRcvAddr(), "data", string(tx.GetData()))
+
+	sw := sc.monitorBeginExecuteSmartContractTransaction()
+	returnCode, err := sc.doExecuteSmartContractTransaction(tx, acntSnd, acntDst)
+	sc.monitorEndExecuteSmartContractTransaction(sw, tx, returnCode, err)
+	return returnCode, err
+}
+
+func (sc *scProcessor) monitorBeginExecuteSmartContractTransaction() *core.StopWatch {
+	sw := core.NewStopWatch()
```
Comment (sasurobert) [inline process/smartContract/process.go:177]: This is a little overcomplicated. There is no need for this function

File: process/smartContract/process.go [hunk 2]
```diff
@@ -160,12 +163,42 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
...
+}
+
+func (sc *scProcessor) monitorEndExecuteSmartContractTransaction(sw *core.StopWatch, tx data.TransactionHandler, returnCode vmcommon.ReturnCode, err error) {
+	sw.Stop("execute")
+	duration := sw.GetMeasurement("execute")
+
+	logFunc := log.Trace
+	if duration > executeDurationDebugThreshold {
+		logFunc = log.Debug
+	}
```
Comment (sasurobert) [inline process/smartContract/process.go:189]: Warn is enough.

File: process/smartContract/process.go [hunk 3]
```diff
@@ -160,12 +163,42 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
...
-	defer sc.tempAccounts.CleanTempAccounts()
-
 	if check.IfNil(tx) {
 		return 0, process.ErrNilTransaction
 	}
-	log.Trace("scProcessor.ExecuteSmartContractTransaction()", "sc", tx.GetRcvAddr(), "data", string(tx.GetData()))
+
+	sw := sc.monitorBeginExecuteSmartContractTransaction()
+	returnCode, err := sc.doExecuteSmartContractTransaction(tx, acntSnd, acntDst)
+	sc.monitorEndExecuteSmartContractTransaction(sw, tx, returnCode, err)
```
Comment (sasurobert) [inline process/smartContract/process.go:172]: Do directly here the monitorization and doExecyte. No need for those 2 functions

File: process/smartContract/process.go [hunk 4]
```diff
@@ -27,6 +27,9 @@ var _ process.SmartContractProcessor = (*scProcessor)(nil)
 
 var log = logger.GetOrCreate("process/smartcontract")
 
+const executeDurationDebugThreshold = float64(0.030)
+const executeDurationWarnThreshold = float64(0.100)
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply process/smartContract/process.go:31]: Fixed.

File: process/smartContract/process.go (hunk 1 above)
Comment (andreibancioiu) [reply process/smartContract/process.go:177]: Fixed, removed the extra 2 functions.

File: process/smartContract/process.go (hunk 2 above)
Comment (andreibancioiu) [reply process/smartContract/process.go:189]: Fixed.

File: process/smartContract/process.go (hunk 3 above)
Comment (andreibancioiu) [reply process/smartContract/process.go:172]: Fixed.

File: process/smartContract/process.go [hunk 5]
```diff
@@ -160,12 +162,30 @@ func (sc *scProcessor) ExecuteSmartContractTransaction(
...
+	returnCode, err := sc.doExecuteSmartContractTransaction(tx, acntSnd, acntDst)
+	sw.Stop("execute")
+	duration := sw.GetMeasurement("execute")
+
+	logFunc := log.Trace
+	if duration > executeDurationWarnThreshold {
+		logFunc = log.Warn
+	}
+
+	logFunc("scProcessor.ExecuteSmartContractTransaction()", "sc", tx.GetRcvAddr(), "data", string(tx.GetData()), "duration", duration, "returnCode", returnCode, "err", err)
```
Comment (raduchis) [inline process/smartContract/process.go:180]: interesting +1

**Pull request merged**